File: Q3/Q3.js
let n = parseInt(prompt("Enter the number of rows for the diamond (half):"));

for (let i = 1; i <= n; i++) 
{
    let spaces = " ".repeat(n - i);

    let stars = "*".repeat(2 * i - 1);
    console.log(spaces + stars);
}

for (let i = n - 1; i >= 1; i--) 
{
    let spaces = " ".repeat(n - i);

    let stars = "*".repeat(2 * i - 1);
    console.log(spaces + stars);
}

File: Q3/Q3.py
n = int(input("Enter the number of rows for the diamond (half): "))

for i in range(1, n + 1):

    print(" " * (n - i), end="")

    print("*" * (2 * i - 1))

for i in range(n - 1, 0, -1):
  
    print(" " * (n - i), end="")
  
    print("*" * (2 * i - 1))
